storage/objects/copy_file.go

```diff
@@ -43,6 +43,20 @@ func copyFile(w io.Writer, dstBucket, srcBucket, srcObject string) error {
 	src := client.Bucket(srcBucket).Object(srcObject)
 	dst := client.Bucket(dstBucket).Object(dstObject)
 
+	// Set a generation-match precondition. The request to upload is aborted
+	// if the object's generation number does not match your precondition
+	// criteria. This avoids race conditions and data corruption.
+	// For a dst object that does not yet exist, set the DoesNotExist precondition.
+	dst = dst.If(storage.Conditions{DoesNotExist: true})
+
+	// If the destination object already exists in your bucket, set instead a
```

Comment:
"If the destination object already exists in the bucket, set a generation-match precondition to the object's generation number."